src/mem/largealloc.h

```diff
@@ -88,6 +85,51 @@ namespace snmalloc
      */
     ModArray<NUM_LARGE_CLASSES, MPMCStack<Largeslab, RequiresInit>> large_stack;
 
+  public:
+    using Pal = PAL;
+
+    /**
+     * Pop an allocation from a large-allocation stack.  This is safe to call
+     * concurrently with other acceses.  If there is no large allocation on a
+     * particular stack then this will return `nullptr`.
+     */
+    SNMALLOC_FAST_PATH void* pop_large_stack(size_t large_class)
+    {
+      void* p = large_stack[large_class].pop();
+      if (p != nullptr)
+      {
+        const size_t rsize = bits::one_at_bit(SUPERSLAB_BITS) << large_class;
+        available_large_chunks_in_bytes -= rsize;
+      }
+      return 0;

```

Comment:
Shouldn't this return `p`?

Reply:
Yes.  I honestly have no idea how this passes any tests as written.

Reply:
You probably would only detect this if you are tracking address space usage, this will not recycle any chunks (i.e. 1MiB ranges) of memory,  So tests that go up and down in usage a lot will leak.  Did it pass CI.  I would expect a Windows test to fail with OOM with this bug.  But on your own machine, probably none of the tests will come close to an issue.